src/Microsoft.ML.Data/Training/TrainerBase.cs

```diff
@@ -29,11 +29,11 @@ private protected TrainerBase(IHostEnvironment env, string name)
             Host = env.Register(name);
         }
 
-        IPredictor ITrainer.Train(TrainContext context) => Train(context);
+        IPredictor ITrainer.Train(TrainCatalog context) => Train(context);
 
-        TPredictor ITrainer<TPredictor>.Train(TrainContext context) => Train(context);
+        TPredictor ITrainer<TPredictor>.Train(TrainCatalog context) => Train(context);
```

Comment:
>TrainCatalog [](start = 46, length = 12)

Hmmm... reading through this, it seems like the majority of changes I've seen at least so far have been due to changing `TrainContext`, which as @sfilipi pointed out should not be done. I'll try to re-review again once that's undone, if that's OK?

Reply:
@TomFinley undone!

Reply:
Thanks, starting again now